### PR TITLE
Move `Mutation` under the `Infection\Mutation` namespace

### PR DESCRIPTION
**Pull request by theofidry**

### CODE REVIEW

File: src/Mutator/NodeMutationGenerator.php [hunk 1]
```diff
@@ -82,7 +82,7 @@ public function __construct(
     }
 
     /**
-     * @return Mutation[]
+     * @return \Infection\Mutation\Mutation[]
```
Comment (maks-rafalko) [inline src/Mutator/NodeMutationGenerator.php]: why is it needed? I think should be reverted

File: src/Visitor/MutationsCollectorVisitor.php [hunk 2]
```diff
@@ -46,7 +45,7 @@
 final class MutationsCollectorVisitor extends NodeVisitorAbstract
 {
     /**
-     * @var Mutation[]
+     * @var \Infection\Mutation\Mutation[]
```
Comment (maks-rafalko) [inline src/Visitor/MutationsCollectorVisitor.php]: same

File: src/Visitor/MutationsCollectorVisitor.php [hunk 3]
```diff
@@ -72,7 +71,7 @@ public function leaveNode(Node $node): ?Node
     }
 
     /**
-     * @return Mutation[]
+     * @return \Infection\Mutation\Mutation[]
```
Comment (maks-rafalko) [inline src/Visitor/MutationsCollectorVisitor.php]: same

File: src/Mutator/NodeMutationGenerator.php (hunk 1 above)
Comment (theofidry) [reply src/Mutator/NodeMutationGenerator.php]: Happens occasionally when doing a refactor to change the class namespace with phpStorm

**Pull request merged**